src/main/java/com/google/sps/servlets/getVideosServlet.java

```diff
@@ -0,0 +1,185 @@
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.lang.*;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.sps.data.Video;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import com.google.cloud.secretmanager.v1.AccessSecretVersionResponse;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1.SecretVersionName;
+
+/** 
+* This servlet is used to update the database with well-being related video activity links.
+*/
+@WebServlet("/videoData")
+public class getVideosServlet extends HttpServlet {
+  private static final String baseURL = "https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=50&order=viewCount&q=yoga&type=video&key=";
+  private static final Logger logger = Logger.getLogger(getVideosServlet.class.getName());
+
+  // This method is used to access the api key stored in gcloud secret manager.
+  public String accessSecretVersion(String projectId, String secretId, String versionId) throws IOException {
+    try (SecretManagerServiceClient client = SecretManagerServiceClient.create()){
+      SecretVersionName secretVersionName = SecretVersionName.of(projectId, secretId, versionId);
+
+      // Access the secret version.
+      AccessSecretVersionResponse response = client.accessSecretVersion(secretVersionName);
+
+      // Return the secret payload as a string.
+      return response.getPayload().getData().toStringUtf8();
+    }
+  }
+
+  @Override
+  public void doPut (HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Deletes queries from last doPut so the datastore results can be updated.
+    Query query = new Query("Video");
```

Comment:
A lot of code in this method are very similar to the doPut method in GetGamesServlet. Have you considered moving them into a utility class?